lib/pry/prompt.rb

```diff
@@ -97,10 +98,35 @@ def prompt_name(name)
       end
     end
 
-    add 'default',
-        "The default Pry prompt. Includes information about the current expression \n" \
-        "number, evaluation context, and nesting level, plus a reminder that you're \n" \
-        'using Pry.' do |context, nesting, _pry_, sep|
+    # @return [String]
+    attr_reader :name
+
+    # @return [String]
+    attr_reader :description
+
+    def initialize(name, description, prompt_procs)
+      @name = name
+      @description = description
+      @prompt_procs = prompt_procs
```

Comment:
`@prompt_procs` is a better name than `@value` for sure, but why break API compatibility? we could easily stay backwards compatible if we did two things:

1) `class Prompt < Struct.new(:name, :description, :value)` - this would allow `#[]` to continue to work, so `Pry::Prompt[:simple][:value]`, `Pry::Prompt[:simple][:name]`, etc would still work.

2. use alias_method to allow `value` to be called, print a deprecation message. 

we're going in the direction where when next pry is released, running `pry` will cause an error in `.pryrc` and die.

__edit: it won't die since pry rescues errors in .pryrc and prints a short backtrace, but still not a good ux__

Reply:
maybe we could just do this, not sure:

```ruby
def [](key)
  if %w(name description).include?(key.to_s)
    warn "A prompt is now an instance of `Pry::Prompt` and not a Hash. \n" \
         "Please use `prompt.#{key}` instead."
     public_send(key) 
  elsif key.to_s == 'value'
    warn "A prompt is now an instance of Pry::Prompt and not a Hash. \n" \
          "Please use `prompt.prompt_procs` instead."
     @prompt_procs 
  else
    super # NoMethodError
  end 
end
```